src/classes/BDI_DataImport_API.cls

```diff
@@ -0,0 +1,102 @@
+/*
+    Copyright (c) 2015 Salesforce.com Foundation
+    All rights reserved.
+    
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+    
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of the Salesforce.com Foundation nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+ 
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.com Foundation
+* @date 2015
+* @group API
+* @group-content ../../ApexDocContent/API.htm
+* @description API class that exposes data import functionality.
+*/
+global with sharing class BDI_DataImport_API {
+
+     private class BDIException extends Exception {}
+     
+    /*******************************************************************************************************
+    * @description Imports all DataImport records that have not been previously imported. Their status is changed
+                    to imported, once the process fishishes.
+    * @return ID The ID of the batch job that performs the data import.
+    */
+    global static ID importData() {
+    	Data_Import_Settings__c diSettings = UTIL_CustomSettingsFacade.getDataImportSettings();
```

Comment:
I'm uncomfortable that you are now getting the settings from the facade, rather than the version already stored on the page.  are you 100% sure this will be the same object, and thus you haven't accidentally broken the pages settings?